libyul/AsmPrinter.h

```diff
@@ -101,7 +102,8 @@ class AsmPrinter
 		return formatDebugData(_node.debugData, !isExpression);
 	}
 
-	Dialect const* const m_dialect = nullptr;
+	Mode const m_printingMode;
```

Comment:
Our convention is to explicitly initialize variables of POD types even if it's redundant (like here, because it will always get initialized in constructor).

I usually use `{}` for this (even though I'd prefer the `= x` style otherwise), because it does not draw attention to what specific value we use (which is irrelevant). It only matters that there is a deterministic value.

```suggestion
	Mode const m_printingMode{};
```

Reply:
Alright, makes sense :)